File: src/main/java/com/enderryno/nuclearcraft/interfaces/PluginStructure.java
package com.enderryno.nuclearcraft.interfaces;

import com.enderryno.nuclearcraft.classes.StructureBlock;
import com.enderryno.nuclearcraft.enums.StructureBehaviour;

import java.util.List;

public interface PluginStructure {
    PluginStructure setRequiredInterface(Boolean required);
    PluginStructure setStructureBehaviour(StructureBehaviour behaviour);
    List<StructureBlock<?>> getStructureBlocks();
    List<StructureBlock<?>> getInterfaceBlocks();
    Boolean requiresInterface();
    StructureBehaviour getStructureBehaviour();
}

File: src/main/java/com/enderryno/nuclearcraft/interfaces/PluginStructure.kt
package com.enderryno.nuclearcraft.interfaces

import com.enderryno.nuclearcraft.classes.StructureBlock
import org.bukkit.Location


interface PluginStructure {
    var id: String
    var structureBlocks: MutableList<StructureBlock>
    var requiredInterface: Boolean

    fun instantiateToWorld(locations: List<Location>)
    fun removeStructureFromWorld(locations: List<Location>)

}
